### PR TITLE
fix(partition-plan): failed to migrate historical partition plan data

### LINKED ISSUE #2156: [Bug]: partition plan data migration may be failed
### ODC version

4.2.4

### OB version

not related

### What happened?

user do not set any partition plan or make all partition plan disable.
odc will startup failed.

### What did you expect to happen?

odc can start up succeed~

### How can we reproduce it (as minimally and precisely as possible)?

follow what happened

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by yhilmare**
#### What type of PR is this?

type-bug

#### What this PR does / why we need it:
odc will startup failed when all partition plan configs is disabled~ we may delete these data to work around this issue.
#### Which issue(s) this PR fixes:

Fixes #2156 

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-migrate/src/main/java/com/oceanbase/odc/migrate/jdbc/common/V42417HistoricalPartitionPlanMigrate.java [hunk 1]
```diff
@@ -70,7 +70,9 @@ public class V42417HistoricalPartitionPlanMigrate implements JdbcMigratable {
     @Override
     public void migrate(DataSource dataSource) {
         JdbcTemplate jdbcTemplate = new JdbcTemplate(dataSource);
-        List<PartitionPlanEntityWrapper> partitionPlanWrappers = generatePartitionPlanEntities(jdbcTemplate);
+        List<PartitionPlanEntityWrapper> partitionPlanWrappers = generatePartitionPlanEntities(jdbcTemplate).stream()
+                .filter(w -> CollectionUtils.isNotEmpty(w.getPartitionPlanTableEntityWrappers()))
+                .collect(Collectors.toList());
```
Comment (MarkPotato777) [inline server/odc-migrate/src/main/java/com/oceanbase/odc/migrate/jdbc/common/V42417HistoricalPartitionPlanMigrate.java:75]: What's the impact of filtering out these disabled plans? Will these plans still exist in ODC 4.2.4?

Comment (yhilmare) [reply server/odc-migrate/src/main/java/com/oceanbase/odc/migrate/jdbc/common/V42417HistoricalPartitionPlanMigrate.java:75]: > What's the impact of filtering out these disabled plans?

nothing, the way we cancel a partition plan is to startup a new partition plan and this new partition plan will contain no table or contain tables which is all disabled~

we will encounter a npe if we do not filter out these partition plans.

> Will these plans still exist in ODC 4.2.4

no, we will filter out these partition plan cause that these partition plans will do nothing and there is no necessary to migrate them~

Comment (MarkPotato777) [reply server/odc-migrate/src/main/java/com/oceanbase/odc/migrate/jdbc/common/V42417HistoricalPartitionPlanMigrate.java:75]: So if users disabled a partition plan before V4.2.4, he/she cannot see that disabled plan any more after V4.2.4? Is that what we expected?

Comment (yhilmare) [reply server/odc-migrate/src/main/java/com/oceanbase/odc/migrate/jdbc/common/V42417HistoricalPartitionPlanMigrate.java:75]: no, he/she will will can see these disabled partition plans but he/she will not see the detail of these partition plan cause that we do not migrate these data. ODC will return empty response when he/she attempt to view these data which is expected~

Comment (MarkPotato777) [reply server/odc-migrate/src/main/java/com/oceanbase/odc/migrate/jdbc/common/V42417HistoricalPartitionPlanMigrate.java:75]: okay, that makes sense to me~

**Review by MarkPotato777 — APPROVED** [review]
LGTM

**Pull request merged**